akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala

```diff
@@ -546,10 +546,7 @@ object Source {
    * the laziness and will trigger the factory immediately.
    */
   def lazyFuture[T](create: () => Future[T]): Source[T, NotUsed] =
-    lazySource { () =>
-      val f = create()
-      future(f)
-    }.mapMaterializedValue(_ => NotUsed)
+    single(()).mapAsyncUnordered(1)(_ => create()).withAttributes(DefaultAttributes.lazyFuture)
```

Comment:
single(create).mapAsyncUnordered(1)(_())?

Reply:
We optimize away mapAsync(1) to unordered behind the scenes since that is cheaper, so this just shortcuts directly there.